### DISCUSSION TITLE
Support for batched nvblox motion generation

### DISCUSSION THREAD

**Discussion by sjauhri** [NVlabs/curobo#166] [Research Q&A] [answered]
Hi everyone,
Similar to examples here: https://curobo.org/source/advanced_examples/1_batch_env.html, I wanted to simulate many parallel environments for training while still using the nvblox-based collision checking and motion generation. How could I go about this? Is it even possible to use nvblox in a batched way ?

**Comment by balakumar-s** [comment]
nvblox does provide the option to have many layers, where each layer could be treated as 1 for each robot in batch. However, the depth image to ESDF integration is serial currently. So that would be slow for a batch of environments. How many parallel environments were you thinking?

**Reply by sjauhri** [reply]
I was thinking of about 10 parallel environments to start with. I think Isaac Sim has a limitation of 10 or 16 parallel cameras at the moment

**Reply by balakumar-s** [reply, marked as answer]
For even 2+ environments, nvblox might be slow due to the serial integration. The only workaround is to use mesh or primitives for the world in batched mode.

**Reply by sjauhri** [reply]
Alright, thanks

**Reply by sjauhri** [reply]
Hi @balakumar-s,

Another question related to this. Is it currently possible to use the mpc/mppi solver in batched mode?
A brief look at the MpcSolver class suggests this should be possible
